staking/cli/src/instructions.rs

```diff
@@ -1191,3 +1205,301 @@ pub async fn save_stake_accounts_snapshot(rpc_client: &RpcClient) {
         .unwrap();
     }
 }
+
+
+pub async fn fetch_delegation_record(
+    rpc_client: &RpcClient,
+    positions_address: &Pubkey,
+    publisher: &Pubkey,
+) -> Option<DelegationRecord> {
+    let delegation_record_key = get_delegation_record_address(*publisher, *positions_address);
+
+    let response = rpc_client.get_account_data(&delegation_record_key).await;
+
+    match response {
+        Ok(data) => Some(DelegationRecord::try_deserialize(&mut data.as_slice()).unwrap()),
+        Err(err) => {
+            if err.to_string().contains("AccountNotFound") {
+                None
+            } else {
+                panic!("Error fetching delegation record: {:?}", err);
+            }
+        }
+    }
+}
+
+
+pub async fn advance_delegation_records<'a>(
+    rpc_client: &RpcClient,
+    signer: &dyn Signer,
+    positions: &DynamicPositionArray<'a>,
+    current_epoch: u64,
+    pool_data: &PoolData,
+    pool_data_address: &Pubkey,
+    pyth_token_mint: &Pubkey,
+    pool_config: &Pubkey,
+) -> bool {
+    let positions_address = positions.acc_info.key;
+
+    // Find all publishers the stake account has exposure to
+    let position_account_delegates: Vec<_> = pool_data
+        .publishers
+        .iter()
+        .enumerate()
+        .filter_map(|(publisher_index, publisher)| {
+            if *publisher == Pubkey::default() {
+                return None;
+            }
+
+            let publisher_exposure = {
+                let mut publisher_exposure = 0;
+                for i in 0..positions.get_position_capacity() {
+                    if let Some(position) = positions.read_position(i).unwrap() {
+                        if (position.target_with_parameters
+                            == TargetWithParameters::IntegrityPool {
+                                publisher: *publisher,
+                            })
+                        {
+                            publisher_exposure += position.amount;
+                        }
+                    }
+                }
```

Comment:
this can cause issues if someone unstake but don't claim rewards

Reply:
they can't clean up their positions until they claim everything